L8-L9-Callections/L8-Zad.2-OwnList/src/main/java/pl/babiarz/CustomList.java

```diff
@@ -0,0 +1,69 @@
+package pl.babiarz;
+
+import java.util.*;
+
+public class CustomList implements OwnList {
+    Integer[] customList = new Integer[5];
+
+    @Override
+    public int size() {
+        return customList.length;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        for (Integer integer : customList) {
+            if (integer == null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    @Override
+    public void add(Integer element) {
+        for (int i = 0; i < customList.length; i++) {
+            if (i == customList.length - 1 && customList[i] != null) {
+                customList = Arrays.copyOf(customList, customList.length + 1);
+                customList[customList.length - 1] = element;
+                break;
+            }else if (customList[i] == null) {
+                customList[i] = element;
+                break;
+            }
```

Comment:
A co jeżeli chcę dodać do listy 1000 elementow? A co jeżeli 1 000 000? ;)

Reply:
Podpowiedź:

Lista powinna być dynamicznie rozszerzalna (zarówno mieć możliwość trzymania 1 elemtnu jak i nieskończonej ilosci elementów)

Zacznij od np. 5 elementów, w przypadku gdy ktoś dodaje element 6 - musisz stworzyć nową większą tablicę i przepisać poprzednią + ten nowy element.

Przyda się pewnie jakaś zmienna, która będzie Ci mówiła ile aktualnie elementów jest w liście. Dzięki temu metodę isEmpty będziesz mógł uprościć do:

`return 'length == 0'`

Reply:
A nie robię tego o czym mówisz z rozszerzaniem tablicy w linijkach 27-29? Sprawdziłem to, można dodać 5 elementów, potem szósty i tablica rozszerza się do 6, kolejny, do 7 itd.